### PR TITLE
Refactor registry plugin and simplify its usage

### PR DESCRIPTION
**Pull request by kezhenxu94**
## Purpose of the pull request

This pull request refactors the registry plugin mechanism and simplifies its usage and registry plugin development.

Now the configuration to run locally and in production is the same so users won't get errors like https://github.com/apache/dolphinscheduler/issues/6129 https://github.com/apache/dolphinscheduler/issues/5643 and https://github.com/apache/dolphinscheduler/issues/5793

## Brief change log

- Split the registry plugin from DolphinScheduler's plugin system and use Java SPI directly.
- Minimize the interface of registry so contributors can easily develop a new registry plugin.

### CODE REVIEW

**Comment by kezhenxu94** [comment]
@CalvinKirs @dailidong @lenboo can you please take a look?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -15,8 +15,6 @@
 # limitations under the License.
 #
 
-# This file is only to override the production configurations in standalone server.
-
-registry.plugin.dir=./dolphinscheduler-dist/target/dolphinscheduler-dist-2.0.0-SNAPSHOT/lib/plugin/registry/zookeeper
-registry.plugin.name=zookeeper
-registry.servers=127.0.0.1:2181
+dolphinscheduler/dolphinscheduler-e2e @kezhenxu94
```
Comment (lenboo) [inline .github/CODEOWNERS:18]: is it necessary?  '@kezhenxu94'

Comment (kezhenxu94) [reply .github/CODEOWNERS:18]: > is it necessary? '@kezhenxu94'

This is an example to list "maintainers" for some specific modules, github will automatically request their review when these modules are modified in pull requests, you and other maintainers can also add the modules you are responsible for. Does it make sense to you?

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Comment by davidzollo** [comment]
good job

**Pull request merged**